### PR TITLE
95 disable local ip address for android build

### LINKED ISSUE #95: Disable local-ip-address on Android
@mark-prins mentioned that android breaks with local-ip-address. It's used in mDNS discovery, can be removed for android feature

### PR DESCRIPTION
**Pull request by andreievg**
closes #95 

Was a bit more involved than i hoped.

* For some reason android flag was on (due to default features, so changed to android feature)
* Changed tomls to be explicit about features
* Disabled discovery with android feature

I tried building with the following and saw a bunch of warnings, and some compilation errors, so did a little refactor to remove all compilation warnings

`cargo build --features android --release && cargo build --features postgres --release && cargo build --release`

### CODE REVIEW

File: server/android/makefile [hunk 1]
```diff
@@ -31,4 +31,4 @@ android: $(ARCHS_ANDROID)
 
 .PHONY: $(ARCHS_ANDROID)
 $(ARCHS_ANDROID): %:
-	$(CARGO_LINKER_ENV_$@)="$(CC_$@)" CC="$(CC_$@)" AR="$(AR_$@)" cargo build --target $@ --release
+	$(CARGO_LINKER_ENV_$@)="$(CC_$@)" CC="$(CC_$@)" AR="$(AR_$@)" cargo build --features android --target $@ --release
```
Comment (mark-prins) [inline server/android/makefile:34]: interesting - I'm still getting this when running `make android`

Screen Shot 2022-06-08 at 9 22 58 AM

**Review by mark-prins — COMMENTED** [review]
Changes look good to me - thanks!

I'm having issues still with compilation, though, so am trying things out on the build server..
Unable to compile directly ( `error: could not compile `server`` ) using cargo and unable to make using `make android` ( same error as yesterday ).

have tried using platform [specific dependencies](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#platform-specific-dependencies) too  such as `[target.'cfg(not(target_arch = "aarch64"))'.dependencies]` - no luck with that either yet.

**Comment by andreievg** [comment]
Sorry @mark-prins, i should have done an extensive test (by fully recompiling and searching output for local-ip-address), should be ok now: e20aaa90679f0173a24c08b5467121c71a0ced03
[out.txt.zip](https://github.com/openmsupply/open-msupply/files/8857149/out.txt.zip)

File: server/repository/src/database_settings.rs [hunk 2]
```diff
@@ -1,8 +1,4 @@
-use diesel::{
-    connection::SimpleConnection,
-    r2d2::{ConnectionManager, Pool},
-    SqliteConnection,
-};
+use diesel::r2d2::{ConnectionManager, Pool};
```
[the remark this answers is not in the document]
Comment (andreievg) [reply server/repository/src/database_settings.rs:1]: Yeah, those two are common to both features used in the file and the specific imports are:
https://github.com/openmsupply/open-msupply/blob/93fe15074bbdcef25edc5fd7eadafb22eeaf9e29/server/repository/src/database_settings.rs#L77

and https://github.com/openmsupply/open-msupply/blob/93fe15074bbdcef25edc5fd7eadafb22eeaf9e29/server/repository/src/database_settings.rs#L76 (diesel:SqliteConnection)

You'll notice that with first permalink `SimpleConnection` doesn't look like it's used, but it's a trait that is being used indirectly

**Review by mark-prins — APPROVED** [review]
🎉  Nothing to add ...except thanks!!

**Pull request merged**